embedded/controlsWrapper.h

```diff
@@ -0,0 +1,28 @@
+#ifndef CONTROLSWRAPPER_H
+#define CONTROLSWRAPPER_H
+
+#include <QObject>
+#include <vector>
+#include <unistd.h>
+#include <QMutex>
+
+class controlsWrapper : public QObject
+{
+    public:
+        explicit controlsWrapper(QByteArray &bytes, std::vector<std::string> &names, std::vector<std::string> &types, QObject *parent = nullptr);
+    public slots: 
+        void startThread();
+        // need other slot for a signal
+    signals:
+        void dataReady();
+    private:
+        void addFloatToArray(QByteArray &dataArr, float data);
+
+        QByteArray &bytes;
+        QMutex mutex;
```

Comment:
I think we also want the mutex to be passed in as a reference, similar to they byte array. Software will also need to access the same mutex, so it will need to be initialized outside of the class